src/main/java/net/hydromatic/lookml/ObjectHandler.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Licensed to the LookML Authors under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this
+ * work for additional information regarding copyright ownership.
+ * The LookML Authors license this file to you under the Apache
+ * License, Version 2.0 (the "License"); you may not use this
+ * file except in compliance with the License.  You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied.  See the License for the specific
+ * language governing permissions and limitations under the
+ * License.
+ */
+package net.hydromatic.lookml;
+
+import java.util.function.Consumer;
+
+/** Handles events that occur while parsing a LookML object.
+ *
+ * @see ListHandler */
+public interface ObjectHandler {
+  /** Adds a numeric property. */
+  ObjectHandler number(String propertyName, Number value);
```

Comment:
Are you saying that
```
  "my_prop": 1
```
is valid LookML and equivalent to
```
  my_prop: 1
```
Or that it should be?

Instead of wrapping (`value`, `isLiteral`) in an `PropertyName` class, we could expand the `String propertyName` method parameter to two parameters. There are pros and cons to this 'flattening'. (We are doing something similar with position in #14.)

Should we apply the same treatment to the other places that identifiers can occur - property values (e.g. `x: "true"`), object names (e.g. `x: "abc" {}), and values in lists (e.g. `x: ["abc", "def"]`).

Reply:
Yes.  ` "my_prop": 1` and `my_prop: 1` are currently equivalent and both will return `my_prop` as a keyword token